extensions/amp-mathml/1.0/component.js

```diff
@@ -0,0 +1,157 @@
+
+  useEffect(() => {
+    if ((allowLayoutShift || allowDimensions) && dimensions) {
+      onLoadRef.current?.();
+      requestResizeRef.current?.(dimensions);
+    }
+  }, [
+    onLoadRef,
+    requestResizeRef,
+    allowLayoutShift,
+    allowDimensions,
+    dimensions,
+  ]);
+
+  const messageHandler = useCallback(
+    (e) => {
+      const data = tryParseJson(e.data) ?? deserializeMessage(e.data);
+      if (data['type'] == MessageType.EMBED_SIZE) {
+        const {height, width: w} = data;
+        const width = inline ? w : undefined;
+        setDimensions({height, width});
+      }
+    },
+    [inline]
+  );
+
+  let observerCb;
+  const ioCallback = useCallback(
+    (entry) => {
+      const {isIntersecting} = entry;
+      if (!isIntersecting) {
+        setAllowDimensions(true);
+        // unobserve element once it's rendered
+        observerCb(null);
+      }
+    },
+    [setAllowDimensions, observerCb]
+  );
+  observerCb = useIntersectionObserver(ioCallback);
+  // Need to create custom callback ref because ProxyIframeEmbed uses an imperative handle with an property for the node.
+  const observerCbRef = (proxyIframeEmbedHandle) => {
+    const {node: iframeNode} = proxyIframeEmbedHandle;
+    // Observe grandparent div instead of iframe because iframe keeps changing height
+    observerCb(iframeNode?.parentNode.parentNode);
+  };
+
+  const styles = useMemo(() => {
+    const dimensionStyles =
+      (allowLayoutShift || allowDimensions) &&
+      dimensions &&
+      pxDimensions(dimensions);
+
+    return {
+      ...dimensionStyles,
+      ...(inline ? INLINE_STYLES : {}),
+      ...bentoMathmlStyles,
+    };
+  }, [
+    allowLayoutShift,
+    allowDimensions,
+    dimensions,
+    inline,
+    bentoMathmlStyles,
+  ]);
+
+  const iframeOptions = {formula};
+
+  return (
+    <ProxyIframeEmbed
+      allowFullscreen
+      ref={useMergeRefs([ref, observerCbRef])}
+      type={TYPE}
+      title={title}
+      options={iframeOptions}
+      matchesMessagingOrigin={ORIGIN_MATCHER}
+      messageHandler={messageHandler}
+      style={styles}
+      {...rest}
+    ></ProxyIframeEmbed>
+  );
+}
+
+const BentoMathml = forwardRef(BentoMathmlWithRef);
+BentoMathml.displayName = 'Mathml'; // Make findable for tests.
+export {BentoMathml};
+
+/**
+ * Takes object with height and width number values
+ * and returns object with pixel string values.
+ * If height and width are invalid, returns undefined.
+ * @param {Object} dimensions
+ * @return {Object}
+ */
+function pxDimensions(dimensions) {
+  const {height, width} = dimensions || {};
+  let dims;
+  if (isDefined(height)) {
+    dims = dims || {};
+    dims.height = px(height);
+  }
+  if (isDefined(width)) {
+    dims = dims || {};
+    dims.width = px(width);
+  }
+  return dims;
+}
+
+/**
+ * @param {*} v
+ * @return {boolean} Returns if the `v` is neither null nor undefined
+ */
+function isDefined(v) {
+  return !(v === null || v === undefined);
+}
```

Comment:
You should be able to `v != null` instead of `isDefined(v)` -- `null == undefined // true`

Reply:
my personal preference is to always prefer triple-equals because i know it can be hard for people (mostly me) to remember the edge cases of the [js equality table](https://eslint.org/docs/rules/eqeqeq). But checking `v != null` for whether `v` is defined is a neat trick

in any case I can remove all this because i'm removing pxDimensions